src/selectedtests/app/controllers/project_task_mappings_controller.py

```diff
@@ -1,130 +1,106 @@
+                        Regex that will be used to decide what build variants are analyzed.
+                        Compares to the build variant's display name
+                        """,
+    )
+
+
+class TaskMappingsResponse(BaseModel):
+    """Model for task mapping responses."""
 
-def add_project_task_mappings_endpoints(
-    api: Api, mongo: MongoWrapper, evg_api: EvergreenApi
-) -> None:
+    task_mappings: List = []
+
+
+@router.get(path="", response_model=TaskMappingsResponse)
+def get(
+    threshold: Decimal = Decimal(0),
+    project: Project = Depends(retrieve_evergreen_project),
+    changed_files: List[str] = Depends(changed_files_parser),
+    db: MongoWrapper = Depends(get_db),
+) -> TaskMappingsResponse:
     """
-    Add to the given app instance the task mapping jobs endpoints of the service.
+    Get a list of correlated task mappings for an input list of changed source files.
 
-    :param api: An instance of a Flask Restplus Api that wraps a Flask instance
-    :param mongo: Mongo Wrapper instance
-    :param evg_api: An instance of the evg_api client
+    :param db:
+    :param project: The evergreen project.
+    :param changed_files: List of source files to calculate correlated tasks for.
+    :param threshold: Minimum threshold desired for flip_count / source_file_seen_count ratio
     """
-    ns = api.namespace("projects", description="Project mappings")
-
-    task_mappings_work_item = ns.model(
-        "TaskMappingsWorkItem",
-        {
-            "source_file_regex": fields.String(
-                description="Regex describing folder containing source files in given project",
-                required=True,
-            ),
-            "module": fields.String(description="Module to include in the analysis"),
-            "module_source_file_regex": fields.String(
-                description="""
-                            Regex describing folder containing source files in given module.
-                            Required if module param is provided.
-                            """
-            ),
-            "build_variant_regex": fields.String(
-                description="Regex that will be used to decide what build variants are analyzed. "
-                "Compares to the build variant's display name"
-            ),
-        },
+    task_mappings = get_correlated_task_mappings(
+        db.task_mappings(), changed_files, project.identifier, threshold
     )
+    return TaskMappingsResponse(task_mappings=task_mappings)
 
-    parser = reqparse.RequestParser()
-    parser.add_argument(
-        "changed_files",
-        location="args",
-        help="List of source files to calculate correlated tasks for",
-        required=True,
-    )
-    parser.add_argument(
-        "threshold",
-        type=Decimal,
-        location="args",
-        help="Minimum threshold desired for flip_count / source_file_seen_count ratio",
+
+@router.post(path="", response_model=CustomResponse)
+def post(
```

Comment:
The 404 and 422 responses for the post methods are not showing up in swagger

Reply:
Good catch.